### PR TITLE
1.5 dataframe changes

### PR DESCRIPTION
**Pull request by bashtage**
- [x] xref #227
- [x] Tests added: Please use `assert_type()` to assert the type of any return value

### CODE REVIEW

**Review by Dr-Irv — CHANGES_REQUESTED** [review]

File: pandas-stubs/core/frame.pyi [hunk 1]
```diff
@@ -756,6 +761,8 @@ class DataFrame(NDFrame, OpsMixin):
         col_fill: Hashable = ...,
         *,
         inplace: Literal[True],
+        allow_duplicates: _bool = ...,
+        names: Hashable | list[HashableT] = ...,
```
[the remark this answers is not in the document]
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:765]: `names` are column names, so can be valid column names. Everything except `None` is OK from what I can see.

```python

In [12]: pd.DataFrame([[1],[2],[3]],columns=["a"]).reset_index().set_index("a").reset_index(names=[("a",)])
Out[12]:
   (a,)  index
0     1      0
1     2      1
2     3      2
```

[the remark this answers is not in the document]
Comment (bashtage) [reply pandas-stubs/core/frame.pyi:765]: A lot of docs make assumptions about names types (e.g., strings) that do not actually coincide with pandas actual requirements.

[the remark this answers is not in the document]
Comment (Dr-Irv) [reply pandas-stubs/core/frame.pyi:765]: OK

**Review by Dr-Irv — APPROVED** [review]
thanks @bashtage

**Pull request merged**